### DISCUSSION TITLE
General Discussion

### DISCUSSION THREAD

**Discussion by Joshua-Riek** [Joshua-Riek/ubuntu-rockchip#9] [General]
Chat about anything related to the Orange Pi 5 or this project!

**Comment by develperbayman** [comment]
question dont wanna reopen the issue cause im sure its just waydroid being finicky  but did you just install it and run it?
 im getting gbinder service started then it hangs ....looks like its running but its not  .....waydroid is super important to me i would just use android but it does not support tethering so its a no go ......thanks for all your help

well not the same errors ...but errors i cant tell if its a waydroid issue or whatnot

******log******

(004343) [20:30:58] % mkdir -p /dev/binderfs
(004343) [20:30:58] % mount -t binder binder /dev/binderfs
(004343) [20:30:58] % ln -s /dev/binderfs/anbox-hwbinder /dev/binderfs/anbox-vndbinder /dev/binderfs/anbox-binder /dev/binderfs/vndbinder /dev/binderfs/hwbinder /dev/binderfs/binder /dev/binderfs/binder-control /dev/
(004343) [20:30:58] Save config: /var/lib/waydroid/waydroid.cfg
(004343) [20:30:59] % mkdir -p /var/lib/waydroid/cache_http
(004343) [20:30:59] Downloading https://sourceforge.net/projects/waydroid/files/images/system/lineage/waydroid_arm64/lineage-18.1-20230211-GAPPS-waydroid_arm64-system.zip/download
(004343) [20:32:42] Validating system image
(004343) [20:32:42] Extracting to /var/lib/waydroid/images
(004343) [20:33:32] Save config: /var/lib/waydroid/waydroid.cfg
(004343) [20:33:33] Downloading https://sourceforge.net/projects/waydroid/files/images/vendor/waydroid_arm64/lineage-18.1-20230212-MAINLINE-waydroid_arm64-vendor.zip/download
(004343) [20:33:41] Validating vendor image
(004343) [20:33:41] Extracting to /var/lib/waydroid/images
(004343) [20:33:43] Save config: /var/lib/waydroid/waydroid.cfg
(004343) [20:34:09] % lxc-info --version
5.0.0~git2209-g5a7b9ce67
(004343) [20:34:24] % mkdir -p /var/lib/waydroid/lxc/waydroid
(004343) [20:34:24] % sh -c cat "/usr/lib/waydroid/data/configs/config_base" "/usr/lib/waydroid/data/configs/config_3" "/usr/lib/waydroid/data/configs/config_4" > "/var/lib/waydroid/lxc/waydroid/config"
(004343) [20:35:05] % sed -i s/LXCARCH/aarch64/ /var/lib/waydroid/lxc/waydroid/config
(004343) [20:35:08] % cp -fpr /usr/lib/waydroid/data/configs/waydroid.seccomp /var/lib/waydroid/lxc/waydroid/waydroid.seccomp
(004343) [20:35:15] % aa-enabled --quiet
(004343) [20:35:26] % systemctl is-active -q apparmor
(004343) [20:35:38] % mv /var/lib/waydroid/config_nodes /var/lib/waydroid/lxc/waydroid
(001062) [20:35:40] % lxc-info -P /var/lib/waydroid/lxc -n waydroid -sH
STOPPED
(001062) [20:36:01] % chmod 666 -R /dev/binder
(001062) [20:36:01] % chmod 666 -R /dev/vndbinder
(001062) [20:36:01] % chmod 666 -R /dev/hwbinder
(004439) [20:37:47] Starting waydroid session
(001062) [20:37:47] % /usr/lib/waydroid/data/scripts/waydroid-net.sh start
vnic is waydroid0
(001062) [20:37:48] % systemctl is-active -q nfcd
(001062) [20:37:48] % chmod 777 -R /dev/ashmem
(001062) [20:37:48] % chmod 777 -R /dev/sw_sync
(001062) [20:37:48] % chmod 777 -R /sys/kernel/debug/sync/sw_sync
(001062) [20:37:48] % chmod 777 -R /dev/dri
(001062) [20:37:48] % chmod 777 -R /dev/fb0
(001062) [20:37:48] % chmod 777 -R /dev/video-enc0
(001062) [20:37:48] % chmod 777 -R /dev/video-dec0
(001062) [20:37:48] % mv /var/lib/waydroid/config_session /var/lib/waydroid/lxc/waydroid
(001062) [20:37:48] % mount -o ro /var/lib/waydroid/images/system.img /var/lib/waydroid/rootfs
(001062) [20:37:48] % mkdir -p /var/lib/waydroid/overlay_work/system
(001062) [20:37:48] % mount -t overlay -o ro,lowerdir=/var/lib/waydroid/overlay:/var/lib/waydroid/rootfs,upperdir=/var/lib/waydroid/overlay_rw/system,workdir=/var/lib/waydroid/overlay_work/system,xino=off overlay /var/lib/waydroid/rootfs
(001062) [20:37:48] % mount -o ro /var/lib/waydroid/images/vendor.img /var/lib/waydroid/rootfs/vendor
(001062) [20:37:48] % mkdir -p /var/lib/waydroid/overlay_work/vendor
(001062) [20:37:48] % mount -t overlay -o ro,lowerdir=/var/lib/waydroid/overlay/vendor:/var/lib/waydroid/rootfs/vendor,upperdir=/var/lib/waydroid/overlay_rw/vendor,workdir=/var/lib/waydroid/overlay_work/vendor,xino=off overlay /var/lib/waydroid/rootfs/vendor
(001062) [20:37:48] % mount -o bind /var/lib/waydroid/waydroid.prop /var/lib/waydroid/rootfs/vendor/waydroid.prop
(001062) [20:37:48] Save config: /var/lib/waydroid/waydroid.cfg
(001062) [20:37:48] % lxc-start -P /var/lib/waydroid/lxc -F -n waydroid -- /init
(001062) [20:37:48] New background process: pid=4514, output=background
(001062) [20:37:48] % lxc-info -P /var/lib/waydroid/lxc -n waydroid -sH
lxc-start: waydroid: cgroups/cgroup2_devices.c: bpf_program_load_kernel: 332 Operation not permitted - Failed to load bpf program: (null)
RUNNING
(004439) [20:37:48] Failed to start Clipboard manager service, check logs
(006649) [20:51:10] Starting waydroid session
(006649) [20:51:10] org.freedesktop.DBus.Python.RuntimeError: Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/dbus/service.py", line 715, in _message_cb
    retval = candidate_method(self, *args, **keywords)
  File "/usr/lib/waydroid/tools/actions/container_manager.py", line 34, in Start
    do_start(self.args, session)
  File "/usr/lib/waydroid/tools/actions/container_manager.py", line 133, in do_start
    raise RuntimeError("Already tracking a session")
RuntimeError: Already tracking a session

(006649) [20:51:10] RuntimeError: Already tracking a session

**Reply by Joshua-Riek** [reply]
I initially tested to ensure it was installed correctly and did not run waydroid. But I do have it running.
It was acting strange during the installation, and I think it might be an issue with hardware acceleration. I'll open up the issue again, and we can continue the discussion there. I'm going to do a fresh Ubuntu install and write my exact steps down, hopefully, you can the same result! 
![Screenshot from 2023-02-12 03-40-09](https://user-images.githubusercontent.com/10427125/218291896-d5ba288c-973a-47d4-b605-9dc5c3ce6ed8.png)

**Reply by develperbayman** [reply]
nice!

**Comment by ypatriot** [comment]
I try your 1.4 release image, first boot it has some error, but after reboot, it's amazing, the the chromium video is smooth. i try to test glmark2-es2-wayland I got 4657, thankyou very much

**Reply by Joshua-Riek** [reply]
Hmm, do you know what error was shown? Or did the system hang? It's probably because the filesystem resizes to fill the SD card on the first boot. I want to come up with a more reliable solution.

**Reply by ypatriot** [reply]
at first boot (SDCARD)the system not hang, it keep some process, it seem like file system resize as you said and i let it about 3 minutes and i force to reboot then the system boot well.
at second time I flashed the image to the nvme-ssd via USB adapter, the system boot from USB without some errors at firstboot and after reboot it boot succesfully. 
when i put nvme directly to the pcie the system can not boot from nvme, how to resolve this, thankyou

**Reply by Joshua-Riek** [reply]
I have not had time to look into USB or NVME booting, currently, the system is designed to only boot from an SDCARD.
I would like to get USB and NVME booting to work sooner rather than later, so it's something ill put on the top of my list of things to do. I expect by the end of the week ill have it working, or an update.

**Reply by ypatriot** [reply]
Anyway I Will wait for any good update, thanks very much

**Reply by Joshua-Riek** [reply]
Hey! I just uploaded a new release [v1.5](https://github.com/Joshua-Riek/ubuntu-orange-pi5/releases/tag/v1.5) that fixes the issue with NVME booting, let me know if there are any more issues.

**Reply by ypatriot** [reply]
Thanks for your best efforts, I Will try it and report the result

**Reply by ypatriot** [reply]
Helo, I tested the v1.5 image, flash from opi5 to USB-NVME adapter it can not boot,
And reflash from opi5 directly to PCIE-NVME and it boot succesfully, everything is good , I Will report you further

**Reply by Joshua-Riek** [reply]
Interesting, I just tried to boot from my USB-NVME adapter, and it can't boot. 
Thank you for finding this, something I overlooked completely.

**Reply by ypatriot** [reply]
Anyway I enjoy the last version image, thanks

**Reply by Joshua-Riek** [reply]
Btw did you encounter any error on the first boot with the v1.5 image?

**Reply by ypatriot** [reply]
I got no error while booting V1.5, straightly bring me to the login screen

**Reply by Joshua-Riek** [reply]
Awesome, thanks for the information. I also found out why booting from USB-NVME was not working, this will be fixed in the next release.

**Comment by ferrellsl** [comment]
Just wanted to say thank you.  I tried OrangePi Debian and also Armbian and they both had GPU acceleration issues.  Armbian also didn't recognize my USB WiFi adapter and I had to apply some hacks to Armbian to get HDMI audio output.  Your Ubuntu build worked perfectly.

**Reply by Joshua-Riek** [reply]
Thanks! I'm glad a lot of people are finding this project to be helpful!

**Comment by ferrellsl** [comment]
I just wrote your Ubuntu image to my nvme drive using dd but I need to modify the SPI in order to boot from the nvme drive.  Is there a config tool in this build that I can use to do that?  The other distros all included an orangepi5-config tool or an orangepi5-install tool to do just that.  Thanks in advance.

**Reply by Joshua-Riek** [reply]
I do not currently have a tool to do this, but I will be working on that in the future for ease of use! The easiest way to modify the SPI is by booting into Armbian, or the OrangePi Debian build and using their config tool. My version of Ubuntu will work with either.

**Comment by ferrellsl** [comment]
Yes, that works and I had already prepped the SPI when I had tried Armbian and Debian.  When I dd'ed your image to my nvme drive I didn't use gdisk to remove any existing partitions first.  That seems to make a difference because I went back and removed them and dd'ed your image again and my board now successfully boots with your build.  
 

    On Sunday, February 19, 2023 at 11:00:45 AM MST, Joshua Riek ***@***.***> wrote:  
 
 


I do not currently have a tool to do this, but I will be working on that in the future for ease of use! The easiest way to modify the SPI is by booting into Armbian, or the OrangePi Debian build and using their config tool. My version of Ubuntu will work with either.

**Comment by ypatriot** [comment]
Hello, I'm happy to test v1.6 image, it boots with only part time splash screen, I wish it boot entirely with splash without the boot process text that I cannot watch well.
May be it Will be comfortable if I can install new os update without losing previous installed software, how to achieve this

**Reply by Joshua-Riek** [reply]
I forgot to add an option that hides that text in the release, It will be included in the next.
As for updating the OS without losing your installed software, I will need to come up with a process for this. I have a few ideas on how this can be done but It will require some time to develop properly.

**Reply by ypatriot** [reply]
I'm waiting for new update, thanks

**Comment by ferrellsl** [comment]
I'm running your build from my nvme drive which isn't fully supported yet so I've put my question/issue into the Discussion area rather than Issues.  I've noticed an issue when powering off my board from the desktop or the terminal.  My board isn't completely powering off at shutdown because my cooling fan continues to run (red light does go out though) and it won't boot again until I cycle the power by pulling the power supply from the socket.  Sometimes it takes several power cycles before it finally starts the boot process again.  Is anyone else having this problem?

**Reply by Joshua-Riek** [reply]
I have not done a lot of testing with the nvme drive yet, but it sounds like your board is not getting enough power. Do you know the voltage and amperage of the power adapter you are using? For example, my power adapter is rated at 5.1V 3.5A.

**Reply by ferrellsl** [reply]
I'm using the power supply from OrangePi which is 5v at 4 amps.  I didn't experience the power issues under Debian or Armbian.

**Reply by Joshua-Riek** [reply]
Very strange. I have no clue why that is happening, then. I will need to try and reproduce this to investigate further.

**Reply by ferrellsl** [reply]
I suspect it's related to how Ubuntu handles advanced power management (APM) and suspend states because the fan continues to run even after shutdown by using the shutdown -P command in the terminal..

**Reply by Joshua-Riek** [reply]
Does it work fine for you when using an SD card? Or do you have the same issue?

**Reply by ferrellsl** [reply]
Same behavior with the SD card.  Ubuntu shuts down, the red LED turns off, but the cooling fan continues to run indefinitely after shutdown.  It does seem to boot easier though because I don't have to power cycle the board to boot again as long as I'm booting from the SD card.

**Reply by Joshua-Riek** [reply]
I just tested the latest Ubuntu 22.04 release, and when I use the command `shutdown now` my system seems to fully shut down. The fan and led turned off as expected. If you don't have this issue with Armbian or the OrangePi5 os I have no idea what's wrong.

**Reply by ferrellsl** [reply]
OK.  No worries.  it's just a minor nuisance.

**Reply by ferrellsl** [reply]
I think I found the issue. When I was playing with Armbian yesterday I used the orangepi-config tool to play with the CPU governor.  I had it set to 2.0 to 2.4Ghz.  I just set the range back to between 800Mhz and 2.0Ghz and On-Demand instead of Performance settings and I no longer have to power cycle several times to boot from nvme.  The CPU cooling fan still runs after shutdown but I can live with that.

**Reply by Joshua-Riek** [reply]
That's great news! If you find any further issues, please let me know!

**Comment by javiermitchell** [comment]
I wil be installing your last version. To be honest I was quite impressed with the last one I used, I really appreciate the time yo have taken to make such a useful contribution .
I will b trying to install to NVME, so will post my impressions

**Comment by diepquynh** [comment]
Hi, I'm wondering why there are 2 graphics adapters showing here. In official Ubuntu 22.04 GNOME version for OPi5, there's only the latter one
![image](https://user-images.githubusercontent.com/13411810/224467013-a0a81a2a-ba99-46da-8a5f-0288eabf6316.png)

**Reply by Joshua-Riek** [reply]
To be honest, I have no idea, I tested the official Ubuntu 22.04 GNOME version for OPi5, and the performance does not seem as good as my version. For instance running glmark2, glmark2-es2, or glmark2-es2-wayland has higher FPS results on my version. I would assume the official Ubuntu version needs more tweaks or has some sort of missing drivers compared to my own.

**Reply by ferrellsl** [reply]
That's been my experience as well.  When I run glmark2-es2-wayland on your version of Ubuntu, the score is double (around 4500+) what I get on the official Ubuntu build or Armbian build found at orangepi.org.

**Reply by Joshua-Riek** [reply]
I don't know how exactly they implement the GPU drivers, but my implementation runs well, so I don't see this as an issue :)

**Reply by ferrellsl** [reply]
I agree.  My primary reason for buying an SBC with a Mali-610 GPU was for the graphics performance because I want to use the board for retro gaming and the RK3588 boards simply wipe the floor with the competition in performance and price.  The RK3588 performs at around the same level as a Snapdragon 865 device.  I have a Oneplus 8 Pro cell phone (Snapdragon 865+) and bought it a couple years ago when it was a flagship phone for over $800 and now I can get a RK3588 device that runs multiple operating systems for $130 with 16GB of RAM.  RK3588 devices are simply the biggest bang for the buck out there and why some folks are still looking for Raspberry Pi 4's just baffles me as they're more expensive than RK3588 devices and perform horribly in comparison.

**Reply by diepquynh** [reply]
Same thoughts here. Initially I wanted an SBC just to test out game streaming performance, and RPi4 looks overpriced here (around $140 for the base 2GB model). Then I found this OPi5 for the same price but with much better performance and a whole lot of expansion (for instance, the ability to use NVME SSD). Also the development of this board and chipset type grows really quick. That way I can use this board for something like a real desktop PC or as a server for web services (I do run a lot of docker instances)

Agreed, official OPi5 Linux images are really underperforming compared to @Joshua-Riek's build. Thank you for your great contribution to the community

**Reply by Joshua-Riek** [reply]
It's neat to see the official Orange Pi 5 devs add some of my commits to their kernel and build system, they are watching :eyes:

**Comment by diepquynh** [comment]
Just for the reference, I found Chromium patches here, which is currently supporting up to version 109
https://github.com/JeffyCN/meta-rockchip/tree/master/dynamic-layers/recipes-browser/chromium

**Reply by Joshua-Riek** [reply]
I compiled some of those versions with yocto a few weeks ago. I should probably revisit it as I made some recent GPU changes.

**Reply by diepquynh** [reply]
These patches must have something to do with these green videos I assume
![image](https://user-images.githubusercontent.com/13411810/224506070-f43231cb-8998-4f27-b95b-d73f3f046f4b.png)

**Reply by Joshua-Riek** [reply]
The owner of the ppa im using for chromium and mesa said this in discord about them:
![image](https://user-images.githubusercontent.com/10427125/224506179-d0361ea9-05cf-4c6d-8268-0d02e3748c1f.png)

**Reply by diepquynh** [reply]
Now it does work but GPU process sometimes crashes, so this is still experimental
![image](https://user-images.githubusercontent.com/13411810/224506421-594d21ff-fa37-458a-84f3-c390a758ac38.png)

**Reply by diepquynh** [reply]
Looks like it's applying a shim layer on top of Chromium's GLES library to run with Mali blobs, which I tried to do a similar hack by making symbolic links of `libmali-x11` to `/usr/lib/chromium-browser/{libGLESv2.so,libEGL.so}` but then failed miserably (logs say X-server simply crashed)

**Reply by Joshua-Riek** [reply]
I wish that x11 was better supported, most new arm devices I have worked on only have GPU support for wayland. Or have poor x11 support like this device.

**Reply by diepquynh** [reply]
Managed to get the UI freezing issue. Was doing glmark2-es2 with malirun and it's reproducible. Seems like driver issue
```
[   83.638699] mali fb000000.gpu: GPU Fault 0x00c00488 (GPU_SHAREABILITY_FAULT) in AS1 at 0x00000001302532bf
               ASID_VALID: false,  ADDRESS_VALID: true
[   83.638731] mali fb000000.gpu: GPU Fault 0x00c00488 (GPU_SHAREABILITY_FAULT) in AS2 at 0x00000001302532bf
               ASID_VALID: false,  ADDRESS_VALID: true
[   83.638779] mali fb000000.gpu: GPU Fault 0x00c00488 (GPU_SHAREABILITY_FAULT) in AS3 at 0x00000001302532bf
               ASID_VALID: false,  ADDRESS_VALID: true
[   83.641027] mali fb000000.gpu: GPU Fault 0x24c00488 (GPU_SHAREABILITY_FAULT) in AS2 at 0x000000013025643f
               ASID_VALID: false,  ADDRESS_VALID: true
[   83.641158] mali fb000000.gpu: GPU Fault 0x24c00488 (GPU_SHAREABILITY_FAULT) in AS3 at 0x000000013025643f
               ASID_VALID: false,  ADDRESS_VALID: true
```

**Reply by Joshua-Riek** [reply]
This is one of the reasons I'm only supporting wayland, malirun uses libmali for x11, which has countless issues with the GPU drivers.

**Comment by diepquynh** [comment]
Did anyone try to run IntelliJ IDEA Ultimate on v1.9? It runs great here but the only downside is terminal doesn't run, which I use a lot during development phase

**Reply by Joshua-Riek** [reply]
I have never tried any of JetBrains IDE's on arm64, and I'm honestly impressed with how well they run on this device. When using PyCharm I can open a python console but not a terminal.

**Reply by diepquynh** [reply]
I realized that I downloaded default tar.gz, which was shipped for x86_64 perhaps. I just downloaded ARM64 bundle and terminal worked
![image](https://user-images.githubusercontent.com/13411810/225385999-8cd1ae1d-ba5a-4e2b-835e-2988354d600c.png)

**Reply by Joshua-Riek** [reply]
Awesome!

**Comment by diepquynh** [comment]
So far I've found that AV1 and H264 codecs aren't working well on Chromium (green screen videos), so I tried to disable creating rkvdec instances for both of them and it worked, no more green videos. YouTube videos sometimes use AV1 codec but now they all use VP9, which works well all the time

**Reply by diepquynh** [reply]
If these aforementioned codecs needs to be explicitly used, they will use SW codecs as fallback (which, of course, consume more CPU power and frame drops, but still better than a green screen :D)

**Reply by Joshua-Riek** [reply]
Interesting, how did you disable the AV1 and H264 codecs? I see that there was an update to libv4l-rkmpp that support setting codecs. I assume you disabled the codecs similar to the below?
```bash
{
    echo "type=dec"
    echo "codecs=VP8:VP9:H.265"
    echo "max-width=7680"
    echo "max-height=4320"
} > /dev/video-dec0
chown root:video /dev/video-dec0
chmod 0660 /dev/video-dec0
echo enc > /dev/video-enc0
chown root:video /dev/video-enc0
chmod 0660 /dev/video-enc0
```

**Reply by diepquynh** [reply]
Another (not recommended) way is to just use X11 instead of wayland and malirun Chromium. This way everything works (and given that we're running X11 on X11 mali libs, no crashes or hardlocks were found), but performance is substantially worse. You're also stuck with 60hz display option, because everything beyond that will become a laggy mess

**Reply by diepquynh** [reply]
@Joshua-Riek yeah I did like the method you said

**Reply by diepquynh** [reply]
Note that the codec instances we echo'd to the device file is just for Chromium so it should not affect local video playback (tested 4k60 H264 video on VLC and 1080p 75fps H264 streaming on Moonlight)

**Reply by diepquynh** [reply]
Video instantiation time is also faster too. I assume Chromium or YT was really picky on its codec choices. Usually it takes up to 5s to load a single YT video, then when we seek, it starts lagging around, GPU process crashes then video starts playing again

**Reply by Joshua-Riek** [reply]
How did you get VLC to use hardware acceleration? I tried using it a few weeks ago and did not have any luck.

**Reply by diepquynh** [reply]
Just re-checked. It doesn't use HW codecs. So my only valid conclusion is Moonlight decoding

**Reply by diepquynh** [reply]
Or it does? I don't see supported video acceleration option in Preferences (granted that we don't have VA-API libs for our codecs), but decoding a whole 4k video using SW decoder would have spiked CPU usage to oblivion and I don't see that here

**Reply by diepquynh** [reply]
The CPU is just chilling with only one big core busy
![image](https://user-images.githubusercontent.com/13411810/226468143-ad3a449d-4ab5-4719-a438-cbbb2a4c5086.png)
![image](https://user-images.githubusercontent.com/13411810/226468508-308e634b-7c52-455a-a170-be62144f74be.png)

**Reply by Joshua-Riek** [reply]
I know MPV works for video decoding. Maybe check the GPU usage when using VLC?
```bash
echo "$(cat /sys/devices/platform/fb000000.gpu/devfreq/fb000000.gpu/load | cut -d "@" -f 1)%"
```

**Reply by Joshua-Riek** [reply]
Oh wow, so it is working with hardware acceleration.

**Reply by diepquynh** [reply]
For the context, here's with the snippet you sent (I ran it within a while true loop)
![image](https://user-images.githubusercontent.com/13411810/226469284-535ba660-e81f-4495-9fd3-03b894d3b454.png)

**Comment by diepquynh** [comment]
Today I just tested removing `--use-gl=egl` flag from Chromium defaults and I see GPU rendering is much more stable than using aforementioned flag. The only downside is HW decoders aren't working at all. `chrome://gpu` does say `Video decoding` is `Hardware accelerated` but if when you look at `chrome://media-internals`, it falls back to SW decoder immediately with `VDA Error 4`. I wonder what should be done to actually make HW decoder works (haven't looked at Chromium patches yet)
![image](https://user-images.githubusercontent.com/13411810/226717633-82ff94cd-3abb-428e-90d3-b411923b05b1.png)

**Reply by diepquynh** [reply]
The very basic test for GPU rendering performance is to look at webglsamples Aquarium. Using `--use-gl=egl` flag produces screen flashes everywhere while removing the flag allowed best rendering performance so far

**Comment by gbutts** [comment]
I am a bit of a noob at this but I have managed to get v1.9 of your image onto a self-booting 128Gb M.2 NVME drive. It is working very well (Pi benchmark score of 21,909; glmark2 of 1555) and is nice and stable. Thank you so much for your efforts in bring this to the community.

I note that you have released v1.10 and no doubt further versions in the future. Does the `apt upgrade` process effectively bring a lower version on a par with new versions? Or do we need to reflash (the SSD in my case)? 

Additionally, my installation is now prompting me to upgrade to v22.10. I assuming that doing this may break things. Are you planning to use this version for your image any time soon? 

Thank you again for what you are doing for us newbies.

Geoff

**Reply by Joshua-Riek** [reply]
Currently, `apt upgrade` does not update between releases, but I'm actively working on it. I already set up a package repository for such a purpose, and it will take me some time to ensure it's set up correctly. 

As for v22.10 it should not be prompting you for that, things could break. I thought I disabled the notification. Are you getting it on your desktop interface or through the command line? Could you provide me with a screenshot?

**Reply by gbutts** [reply]
Thank you for your response. I'm looking forward to the upgrade repository when it is done :)

I got it via the desktop after it upgraded some packages. Here is the screenshot:

![Orange_Ubuntu](https://user-images.githubusercontent.com/25043105/227747350-7afb087e-491e-4462-9cbb-abe38b7916ea.jpg)

**Reply by Joshua-Riek** [reply]
I see, could you take a look at the file `/etc/update-manager/release-upgrades`, it should say `Prompt=never` at the bottom of it.

```
ubuntu@ubuntu:~$ cat /etc/update-manager/release-upgrades
# Default behavior for the release upgrader.

[DEFAULT]
# Default prompting and upgrade behavior, valid options:
#
#  never  - Never check for, or allow upgrading to, a new release.
#  normal - Check to see if a new release is available.  If more than one new
#           release is found, the release upgrader will attempt to upgrade to
#           the supported release that immediately succeeds the
#           currently-running release.
#  lts    - Check to see if a new LTS release is available.  The upgrader
#           will attempt to upgrade to the first LTS release available after
#           the currently-running one.  Note that if this option is used and
#           the currently-running release is not itself an LTS release the
#           upgrader will assume prompt was meant to be normal.
Prompt=never
```

**Reply by gbutts** [reply]
Unfortunately, no. It has `Prompt=Normal`

![Cursor_and_geoff_orange-pi5___](https://user-images.githubusercontent.com/25043105/227747652-75de9403-5fd6-4fc8-b924-344c441c711e.jpg)

**Reply by Joshua-Riek** [reply]
Oh, that is the problem. I'll look into ensuring that the value is set to `Prompt=never`, but you can set the value equal to `never` on your system to not receive the 22.10 update prompt.

**Reply by gbutts** [reply]
In case it is relevant, I have updated Ubuntu on my Orange Pi by both methods: using the GUI 'Software Updater' program, and via `sudo apt update && upgrade`.

**Reply by gbutts** [reply]
Ok, will do. Thank you

**Comment by diepquynh** [comment]
I have an Intel AX210 card and I want to install it to OPi5's M.2 key M slot, which I have to use a convert adapter as the card is A+E compatible. I found adapters selling online but most of them requires power delivery through USB pins (or it isn't?). Does anyone know which set of pins in the GPIO headers should I plug the USB power in?

**Comment by sangjela** [comment]
Hi. It work well by image . I use SD for boot. It work well.

Q
. Is can boot nvme? 
. Is gpio... Some thony, python, etc work? 

Thanks for your work.

Have a nice day.

**Reply by Joshua-Riek** [reply]
Yes, the system can boot from a nvme drive. To my knowledge, GPIO is limited at the moment, I have included wiringpi with support for this board. I have no idea if it works with python or thony, I have a link to the wiringpi repo below.

https://github.com/orangepi-xunlong/wiringOP/tree/next

**Reply by sangjela** [reply]
Thanks for your reply. Maybe I use https://www.google.com/amp/s/jamesachambers.com/orange-pi-5-ssd-boot-guide/%3famp=1
Or Readme.md .
I worried about I want sometime change to boot by sd or ssd.
And I can use your link for gpio... I will study more for this.
Thanks again.

**Comment by diepquynh** [comment]
@Joshua-Riek is there any reason that you enabled back DMC? Because from my experience, enabling DMC causing significant performance degradation. With 150Mbps streaming using Moonlight, there's significant decoding performance drop (e.g random lags), whereas running DMC at `performance` governor or disabling it just solves the issue

**Reply by diepquynh** [reply]
This was tested on v1.13 btw

**Reply by Joshua-Riek** [reply]
I enabled DMC in the device tree by default some time ago, I can't remember why I did this, to be honest when there is already an overlay for DMC. I did not know DMC could cause performance degradation like that. I will be sure to disable DMC by default and if someone wants to enable it they can use the overlay.

**Reply by diepquynh** [reply]
Honestly, I don't think we need a scaling governor for anything. It would be reasonable when the main power source is battery, which is not the case for most of the time with the OPi5, unless the CPU itself is so badly binned that even a 20W power brick can't handle the power consumption
